app/controllers/doorkeeper/authorizations_controller.rb

```diff
@@ -31,8 +31,14 @@ def destroy
     private
 
     def render_success
-      if skip_authorization? || (matching_token? && pre_auth.client.application.confidential?)
-        redirect_or_render(authorize_response)
+      if skip_authorization? || ((token = matching_token?) && pre_auth.client.application.confidential?)
```

Comment:
I think we have to do some refactoring with method extractions here, now it starts to look complex. WDYT?

Maybe unify the condition and move attributes setup into some method. It will also allow to override them if required in any application (rather than overriding `render_success`)